version: '3'

services:
  web:
    build:
      context: .
      dockerfile: apps/web/Dockerfile
    ports:
      - '5173:5173'
    environment:
      - VITE_API_URL=http://api:3000
    volumes:
      - ./apps/api/src/epub/output:/app/public/epub
      - ./apps/api/src/ocr/output:/app/public/ocr
    depends_on:
      - api-server

  api-server:
    build:
      context: .
      dockerfile: apps/api/Dockerfile.api
    ports:
      - '3000:3000'
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - ./apps/api/src/epub/output:/app/src/epub/output
      - ./apps/api/src/ocr/output:/app/src/ocr/output
    environment:
      - PGHOST=postgres
      - PGUSER=postgres
      - PGPASSWORD=postgres
      - PGDATABASE=flipbook
      - PGPORT=5432
      - PORT=3000

  postgres:
    image: postgres:14
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_DB=flipbook
    ports:
      - '5432:5432'
    volumes:
      - postgres-data:/var/lib/postgresql/data
      - ./apps/api/src/init-db.sql:/docker-entrypoint-initdb.d/init-db.sql
      - ./apps/api/src/db/001_create_epub_core.sql:/docker-entrypoint-initdb.d/001_create_epub_core.sql
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5

volumes:
  postgres-data:
